src/lib/consentHandler/cmp.js

```diff
@@ -0,0 +1,66 @@
+export const GET_CONSENT_DATA = "getConsentData";
+export const GET_VENDOR_CONSENTS = "getVendorConsents";
+export const GET_SITE_CONSENT = "siteHasConsent";
+export const CMP_GET_MSG = '__cmpCall';
+export const CMP_RETURN_MSG = '__cmpReturn';
+export const CMP_FRAME = '__cmpLocator';
+export const CMP_API = '__cmp';
+
+
+/**
+ * Helper to determine GDPR consents.
+ *
+ * If consentConfig.type is 'iab', then it will attempt to call cmp in the following order:
+ *  1. Cmp in the same frame, or top frame.
+ *  2. If safe frame is used, then call cmp thru safe frame.
+ *  3. Look for a cmp frame and call cmp thru messages.
+ *
+ * If consentConfig.type is 'static', then the driver data is expected to be supplied in
+ * driver.consentData.  Usually this means the caller has managed to call cmp on its own.
+ */
+export default class Cmp {
+    constructor() {
+        this.returnMsgName = CMP_RETURN_MSG;
+        this.locatorFrame = CMP_FRAME;
+        this.cmpApi = CMP_API;
+        this.safeframeCall = "cmp";
+    }
+
+    getConsentCmd() {
+        return [[GET_CONSENT_DATA],[GET_SITE_CONSENT]];
+    }
+
+    getSafeFrameData(data) {
+        return (data.cmpCommand === GET_CONSENT_DATA) ?
+            data.vendorConsentData : data.vendorConsents;
```

Comment:
This doesn't handle the new site command

Reply:
once the definitions on handling safeframes is complete I can finish this

Reply:
left a TODO for now